src/utils/maker-colors.js

```diff
@@ -0,0 +1,97 @@
+import chroma from 'chroma-js';
+import { cloneDeep } from 'lodash';
+import { WCAG_CONTRAST_TEXT, DARK_COLOR_LUMINANCE_THRESHOLD, getContrast } from '@square/maker/utils/get-contrast';
+
+const NEUTRAL_RATIOS = {
+	light: {
+		'neutral-0': 0,
+		'neutral-10': 0.05,
+		'neutral-20': 0.155,
+		'neutral-80': 0.527,
+		'neutral-90': 0.9,
+		'neutral-100': 1,
+	},
+	dark: {
+		'neutral-0': 0,
+		'neutral-10': 0.255,
+		'neutral-20': 0.37,
+		'neutral-80': 0.55,
+		'neutral-90': 0.95,
+		'neutral-100': 1,
+	},
+};
+
+const CONTEXTUAL_COLORS = {
+	light: {
+		critical: {
+			fill: '#cd2026',
+			text: '#a82826',
+			subtle: '#f6eceb',
+		},
+		warning: {
+			fill: '#ffbf00',
+			text: '#7e662a',
+			subtle: '#f9eecf',
+		},
+		success: {
+			fill: '#008000',
+			text: '#0a7A06',
+			subtle: '#ebf1eb',
+		},
+
+	},
+	dark: {
+		critical: {
+			fill: '#cd2026',
+			text: '#ff7566',
+		},
+		warning: {
+			fill: '#ffbf00',
+			text: '#ffbf00',
+		},
+		success: {
+			fill: '#008000',
+			text: '#64cc52',
+		},
+	},
+};
+
+/**
+ * @param {String} background
+ * @return {Object}
+ */
+export default function makerColors(background = '#fff') {
```

Comment:
Lauren can correct me if I'm wrong, but I believe the color generation function is used in multiple places around Website, not just in MTheme, so there's value in exposing it as a standalone function from Maker.

Reply:
@kirillzh currently we just use website's `makerThemeColors` utility in `MTheme` components.
@landondurnan I see what you're saying. I considered that approach but felt letting engineers first use `makerColors` to generate the baseline set of Maker Colors before adding their own `theme.colors` customizations would limit the expectation that `MTheme` would do magical contrast transformations to everything in the `theme.colors`. Since we support unlimited custom color props that can be referenced by components but can't anticipate every color that will need contrast checking and what the contrast threshold will be.